ufersa/plataforma-sabia
# Make CurrencyInputField work with FormWizard

## Summary

Resolves #395 

## Relevant technical choices

* It was necessary to wrap the component around `RHF`'s `Controller` so it works with `form.watch`.
* `react-number-format` doesn't have an option to handle field value without formatting, so I did some modifications in a way that the form already receives the value formatted and also sends the value formatted to the API.

## QA Steps

1. Access edit/registration page of a technology;
2. Fill all fields until reaching Costs page;
3. You should be able to register implementation costs, maintenance costs, and funding value successfully;
4. Click **Save** button and take a look at the API request, no errors should happen;
5. Click **Back** button, all values that you filled should be there again.
6. Click **Save** button again, no validation errors should happen, even pattern validation in input.

## Checklist
